Consider all labels by default when deriving security identity

tgraf opened the pull request.

aanm commented on daemon/docker_watcher.go:
```diff
@@ -183,20 +183,13 @@ func (d *Daemon) fetchK8sLabels(dockerLbls map[string]string) (map[string]string
 			log.Warningf("Error while getting kubernetes labels: %s", err)
 		} else if k8sNormalLabels != nil {
 			k8sLbls = labels.Map2Labels(k8sNormalLabels, k8s.LabelSource)
-
```
We still need to store the `io.kubernetes.pod.namespace` here.

tgraf replied:
Fixed

aanm commented on pkg/labels/filter.go:
```diff
@@ -114,18 +114,11 @@ func (cfg *LabelPrefixCfg) Append(l *LabelPrefix) {
-				Source: k8s.LabelSource,
+				Ignore: true,
+				Prefix: "io.kubernetes",
 			},
```
Also ignore `k8s.io/kubernetes/pkg/apis/extensions/DefaultDeploymentUniqueLabelKey` or `"pod-template-hash"`

tgraf committed:
labels: Simplify parsing of valid label prefixes

- Move into pkg/labels/
- No need to take lock

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
labels: Support ignoring label prefix

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
labels: Consider all labels except "io.kubernetes" by default

Instead of whitelisting individual label prefixes by default,
consider all labels except "io.kubernetes*" to avoid including
pod-name hash, timestamps, grace periods, etc.

Signed-off-by: Thomas Graf <thomas@cilium.io>
